src/librbd/operation/MigrateRequest.cc

```diff
@@ -57,15 +57,17 @@ class C_MigrateObject : public C_AsyncObjectThrottle<I> {
   ::SnapContext m_snapc;
   uint64_t m_object_no;
 
-  io::AsyncOperation m_async_op;
+  io::AsyncOperation *m_async_op = nullptr;
 
   void start_async_op() {
     I &image_ctx = this->m_image_ctx;
     assert(image_ctx.owner_lock.is_locked());
     CephContext *cct = image_ctx.cct;
     ldout(cct, 10) << dendl;
 
-    m_async_op.start_op(image_ctx);
+    assert(m_async_op == nullptr);
```

Comment:
Nit: can you just allocate this within the ```if``` block below so that you don't need to delete it if you fall through?

Reply:
@dillaman Wouldn't we create a race window if we called start_op  after checking writes_blocked? Couldn't writes be blocked between our check and start_op?

Reply:
Good point, never mind